source/common/stats/isolated_store_impl.cc

```diff
@@ -34,21 +33,37 @@ IsolatedStoreImpl::IsolatedStoreImpl(SymbolTable& symbol_table)
         return alloc_.makeTextReadout(name, name, StatNameTagVector{});
       }),
       null_counter_(new NullCounterImpl(symbol_table)),
-      null_gauge_(new NullGaugeImpl(symbol_table)),
-      default_scope_(std::make_shared<ScopePrefixer>("", *this)) {}
+      null_gauge_(new NullGaugeImpl(symbol_table)) {}
+
+ScopeSharedPtr IsolatedStoreImpl::rootScope() {
+  if (lazy_default_scope_ == nullptr) {
+    StatNameManagedStorage name_storage("", symbolTable());
```

Comment:
could you help me to understand the lifecycle management of name_storage vs lazy_default_scope_ ?

Reply:
plus, is there a thread-safe issue on lazy_default_scope_?

Reply:
Yes: the scope that's created below copies and owns the StatName.

Remember, StatName is like string_view: APIs make take a string-view arg, and will usually need to copy it into owned storage.

Reply:
what about the thread-safety question?

Reply:
Oh, sorry. IsolatedStoreImpl is not a thread-safe class. There is a thread-safe wrapper in test/server.

Reply:
I will add a comment (but not push it right now, no need to waste the CI cycles as I need to do another push to fix coverage anyway).